name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/MerkurPrivatBankPDFExtractor.java

```diff
@@ -124,4 +152,25 @@ private <T extends Transaction<?>> void addFeesSectionsTransaction(T transaction
                         .assign((t, v) -> processFeeEntries(t, v, type));
 
     }
+
+    private Transaction<AccountTransaction> depositRemovalTransaction(DocumentType type, String regex)
+    {
+        return new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DEPOSIT);
+            return entry;
+        }).section("date", "amount", "type").match(regex).assign((t, v) -> {
+            if ("S".equals(v.get("type")))
+                t.setType(Type.REMOVAL);
+            Map<String, String> context = type.getCurrentContext();
```

Comment:
Hello @ZfT2 
You dont need `Map<String, String> context = type.getCurrentContext();`

add follow like -> [here](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/DkbPDFExtractor.java#L723:L738) <-

`.documentContext("currency", "year") //`

then you can simply use `v.get("currency")` or `v.get("year")` to retrieve the string, as it is already parsed in the document.